lib/puppet/type/archive.rb

```diff
@@ -150,6 +150,16 @@ def should_to_s(value)
     desc 'extract command group (using this option will configure the archive file permisison to 0644 so the user can read the file).'
   end
 
+  newparam(:proxy_type) do
+    desc 'proxy type (none|ftp|http|https)'
+    newvalues(:none, :ftp, :http, :https)
+    defaultto(:none)
+  end
+
+  newparam(:proxy_server) do
+    desc 'proxy address to use when accessing source'
+  end
+
```

Comment:
I think if you leave :proxy_type without any default you can at the end here do:
```ruby
validate
  self[:proxy_type] ||= URI(self[:proxy_server]).scheme if self[:proxy_server]
end
```

This should make it more convenient than forcing user to specify proxy_type every time.